Error CRITICAL: Unhandled Promise Rejection

### What happened?

Debug Console (ctrl+o to close)                                                                          │
│                                                                                                          │
│ ℹ Authenticated via "qwen-oauth".                                                                       │
│ ℹ Error: No changes to apply. The old_string and new_string are identical.                              │
│ ✖ [ERROR] [IDEClient] IDE connection error. The connection was lost unexpectedly. Please try            │
│   reconnecting by running /ide enable                                                                    │
│   fetch failed                                                                                           │
│ ✖ =========================================                                                             │
│   This is an unexpected error. Please file a bug report using the /bug tool.                             │
│   CRITICAL: Unhandled Promise Rejection!                                                                 │
│   =========================================                                                              │
│   Reason: TypeError: fetch failed                                                                        │
│   Stack trace:                                                                                           │
│   TypeError: fetch failed                                                                                │
│       at fetch                                                                                           │
│   (file:///C:/Users/ccc/AppData/Roaming/npm/node_modules/@qwen-code/qwen-code/cli.js:63319:17)         │
│       at process.processTicksAndRejections (node:internal/process/task_queues:105:5)                     │
│       at async                                                                                           │
│   file:///C:/Users/ccc/AppData/Roaming/npm/node_modules/@qwen-code/qwen-code/cli.js:229796:28          │
│       at async StreamableHTTPClientTransport.send                                                        │
│   (file:///C:/Users/ccc/AppData/Roaming/npm/node_modules/@qwen-code/qwen-code/cli.js:228482:28) (x2)  

### What did you expect to happen?

Error fixed.

### Client information


* **CLI Version:** 0.1.1
* **Git Commit:** 92245f0f
* **Session ID:** 3d1137eb-4b97-4d3d-ae96-9d3bde6cc9de
* **Operating System:** win32 v22.16.0
* **Sandbox Environment:** no sandbox
* **Auth Type:** qwen-oauth
* **Model Version:** coder-model
* **Memory Usage:** 176.5 MB
* **IDE Client:** VS Code


### Login information

_No response_

### Anything else we need to know?

_No response_

--- a/packages/core/src/ide/ide-client.test.ts
+++ b/packages/core/src/ide/ide-client.test.ts
@@ -113,7 +113,7 @@ describe('IdeClient', () => {
         'utf8',
       );
       expect(StreamableHTTPClientTransport).toHaveBeenCalledWith(
-        new URL('http://localhost:8080/mcp'),
+        new URL('http://127.0.0.1:8080/mcp'),
         expect.any(Object),
       );
       expect(mockClient.connect).toHaveBeenCalledWith(mockHttpTransport);
@@ -181,7 +181,7 @@ describe('IdeClient', () => {
       await ideClient.connect();
 
       expect(StreamableHTTPClientTransport).toHaveBeenCalledWith(
-        new URL('http://localhost:9090/mcp'),
+        new URL('http://127.0.0.1:9090/mcp'),
         expect.any(Object),
       );
       expect(mockClient.connect).toHaveBeenCalledWith(mockHttpTransport);
@@ -230,7 +230,7 @@ describe('IdeClient', () => {
       await ideClient.connect();
 
       expect(StreamableHTTPClientTransport).toHaveBeenCalledWith(
-        new URL('http://localhost:8080/mcp'),
+        new URL('http://127.0.0.1:8080/mcp'),
         expect.any(Object),
       );
       expect(ideClient.getConnectionStatus().status).toBe(
@@ -665,7 +665,7 @@ describe('IdeClient', () => {
       await ideClient.connect();
 
       expect(StreamableHTTPClientTransport).toHaveBeenCalledWith(
-        new URL('http://localhost:8080/mcp'),
+        new URL('http://127.0.0.1:8080/mcp'),
         expect.objectContaining({
           requestInit: {
             headers: {

--- a/packages/core/src/ide/ide-client.ts
+++ b/packages/core/src/ide/ide-client.ts
@@ -667,10 +667,10 @@ export class IdeClient {
   }
 
   private createProxyAwareFetch() {
-    // ignore proxy for 'localhost' by deafult to allow connecting to the ide mcp server
+    // ignore proxy for '127.0.0.1' by deafult to allow connecting to the ide mcp server
     const existingNoProxy = process.env['NO_PROXY'] || '';
     const agent = new EnvHttpProxyAgent({
-      noProxy: [existingNoProxy, 'localhost'].filter(Boolean).join(','),
+      noProxy: [existingNoProxy, '127.0.0.1'].filter(Boolean).join(','),
     });
     const undiciPromise = import('undici');
     return async (url: string | URL, init?: RequestInit): Promise<Response> => {
@@ -851,5 +851,5 @@ export class IdeClient {
 function getIdeServerHost() {
   const isInContainer =
     fs.existsSync('/.dockerenv') || fs.existsSync('/run/.containerenv');
-  return isInContainer ? 'host.docker.internal' : 'localhost';
+  return isInContainer ? 'host.docker.internal' : '127.0.0.1';
 }

